### PR TITLE
Hosted paid content Amp fixes

### PR DESCRIPTION
**Pull request by Calanthe**
## What does this change?

Fixes for the Amp versions of the paid content pages.
## What is the value of this and can you measure success?

Amp paid content pages look much better now.

## Does this affect other platforms - Amp, Apps, etc?

Yes, Amp

## Screenshots

Before:
screen shot 2016-10-27 at 15 04 18

After:
screen shot 2016-10-27 at 14 56 40

screen shot 2016-10-27 at 14 56 55
## Request for comment

@NataliaLKB @gtrufitt

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
The baseline of the button is shifted one pixel up. Why is the text in the popup not typeset in Guardian Egyptian?

**Comment by NataliaLKB** [comment]
Would you be able to add a test for a hosted content page?

https://github.com/guardian/frontend/blob/master/article/test/ArticleAmpValidityTest.scala

**Comment by jfsoul** [comment]
> Would you be able to add a test for a hosted content pages?

> https://github.com/guardian/frontend/blob/master/article/test/ArticleAmpValidityTest.scala

and https://github.com/guardian/frontend/blob/master/tools/amp-validation/endpoints.js 😄

**Comment by Calanthe** [comment]
@regiskuckaertz You are (as always) right :) I just made 2 fixes as you suggested.

File: common/app/views/fragments/amp/stylesheets/glabs.scala.html [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+.paidfor-meta__label,
+.paidfor-label {
+    display: flex;
+}
+
+.paidfor-input {
+    display: none;
+}
+
+.paidfor-input:not(checked) ~ .popup--paidfor {
```
Comment (regiskuckaertz) [inline common/app/views/fragments/amp/stylesheets/glabs.scala.html]: Does this work? The pseudo-selector is `:checked` and thus its negation is `:not(:checked)`; or is it?

File: common/app/views/fragments/amp/stylesheets/glabs.scala.html [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    border-radius: 4px;
+    font-weight: 700;
+    z-index: 32000;
+}
+
+.popup--paidfor::before {
+    content: '';
+    border: solid transparent;
+    border-bottom-color: #333;
+    border-style: solid;
```
Comment (regiskuckaertz) [inline common/app/views/fragments/amp/stylesheets/glabs.scala.html]: This one can go away, as `border` suffices.

File: common/app/views/fragments/commercial/paidForMeta.scala.html [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
-        <button class="u-button-reset paidfor-label__btn popup__toggle" data-toggle="js-paidfor-popup@dataId">About @inlineSvg("arrow-down", "icon")</button>
-        <div class="popup popup--paidfor is-off js-paidfor-popup@dataId">
-            <h3 class="popup--paidfor__title">Paid content is paid for and controlled by an advertiser and produced by the Guardian Labs team</h3>
-            <a class="popup--paidfor__link" href="@LinkTo("/content-funding")">Learn more about Guardian Labs content @inlineSvg("arrow-right", "icon")</a>
+    @if(request.isAmp) {
+        <div class="paidfor-label paidfor-meta__more has-popup">
+            <input type="checkbox" value="selected" id="paidforAbout" class="paidfor-input">
+            <label for="paidforAbout" class="tooltip-label">About @inlineSvg("arrow-down", "icon")</label>
+            <div class="popup popup--paidfor">
+                <h3 class="popup--paidfor__title">Paid content is paid for and controlled by an advertiser and produced by the Guardian Labs team</h3>
```
Comment (regiskuckaertz) [inline common/app/views/fragments/commercial/paidForMeta.scala.html]: I don't think this should be a heading.

**Review by regiskuckaertz — COMMENTED** [review]
I'll let the AMP team give you final approval, I just noticed a few minor details that are not really important.

File: common/app/views/fragments/amp/stylesheets/glabs.scala.html (hunk 1 above)
Comment (Calanthe) [inline common/app/views/fragments/amp/stylesheets/glabs.scala.html]: It does work with `:not(checked)`

Comment (regiskuckaertz) [inline common/app/views/fragments/amp/stylesheets/glabs.scala.html]: Yeah perhaps in some browsers, but that is not standards compliant. In more zealous browsers that may be interpreted as an element selector, like `:not(div)`.

**Comment by NataliaLKB** [comment]
> I'll let the AMP team give you final approval, I just noticed a few minor details that are not really important.

There is no AMP team 😆 . But as its part of dotcom, the dotcom platform team are responsible for maintaining it.

**Comment by Calanthe** [comment]
Thanks @regiskuckaertz for your great advises :)

**Comment by NataliaLKB** [comment]
Code lgtm 👍 . @regiskuckaertz did an amazing job with the review 🙌 . 

Teamcity has failed, but I don't think it should have. I hope you don't mind, but I have kicked off another one to see for sure.

**Comment by Calanthe** [comment]
Thanks @NataliaLKB ! I don't know why the teamcity build is failing. Amp validation works fine on my local machine.

**Comment by jfsoul** [comment]
@Calanthe I think you'll need to commit the test fixture file that should have been generated when you ran the tests locally.

(cc @NataliaLKB)

**Comment by NataliaLKB** [comment]
@jfsoul found the problem! It is possible that the scala tests are using an older version amp validator and may need to update it. He is investigating now.

**Comment by jfsoul** [comment]
let's see whether teamcity passes now 🍀

**Comment by NataliaLKB** [comment]
Thanks @jfsoul  all looks good now.

**Pull request merged**